From eaa986dd50ef2377826c541d4d2d728ea9eb05ec Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:36:00 -0400
Subject: [alsa] hda/hdmi: call overridden init on resume

Message-id: <140972823573-348-patchset-jkysela@redhat.com>
Patchwork-id: 91044
O-Subject: [RHEL 7.1 PATCH 347/442] ALSA: hda - hdmi: call overridden init on resume
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

We need to call the proper init function in case it has been
overridden, as it might restore things that the generic routing
doesn't know anything about. E.g. AMD cards have special verbs
that need resetting.

Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=77901
Fixes: 5a61358433b1 ('ALSA: hda - hdmi: Add ATI/AMD multi-channel audio support')
Signed-off-by: Pierre Ossman <pierre@ossman.eu>
Cc: <stable@vger.kernel.org> [v3.13+]
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: a283368382c50345dff61525f493ea307f21ec9b)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_hdmi.c b/sound/pci/hda/patch_hdmi.c
index 3e4417b..4fe876b 100644
--- a/sound/pci/hda/patch_hdmi.c
+++ b/sound/pci/hda/patch_hdmi.c
@@ -2204,7 +2204,7 @@ static int generic_hdmi_resume(struct hda_codec *codec)
  struct hdmi_spec *spec = codec->spec;
  int pin_idx;
 
- generic_hdmi_init(codec);
+ codec->patch_ops.init(codec);
  snd_hda_codec_resume_amp(codec);
  snd_hda_codec_resume_cache(codec);
 
-- 
1.7.1